File: r/R/examples/layout.R
# Each line of code has an associated description
# Layout / Position
# How to adjust the positons of cards on a page

#library(h2owave)

# Check if a variable with the name "page_name" exists in an object page. This is a check to remove any previous page objects.
if("page_name" %in% names(page)) page$drop()
# Create a new page called "demo". 
page <- Site("/demo")

# Number of columns on the page.
columns = 12
# Number of rows on the page.
rows = 10

# For-loop to cycle through 12 columns.
for(column in 1:columns){
        # And a nested for-loop to cycle through 10 rows.
        for(row in 1:rows){
                # Multiple cards are added to page "demo". Each card is dynamically placed by the column number and row number. And each card is of column size 1, and row size 1. 
               box = paste0(column," ",row," 1 1") 
        # Each card is named according to its position by pasting the column and row number. 
        page$add_card(paste0("card_",column,"_",row),
                        # Each card is a markdown card.
                              ui_markdown_card(box=box,title=box,content=''))
        }}
# Save the page object. Saving the page object pushes the card object to the wave server. 
page$save()

File: py/examples/layout.py
# Layout / Position
# How to adjust the position of cards on a page.
# #layout
# ---

from h2o_wave import site, ui

# Every page has a grid system in place.
# The grid has 12 columns and 10 rows.
# A column is 134 pixels wide.
# A row is 76 pixels high.
# The gap between rows and columns is set to 15 pixels.

# Cards have a `box` attribute that specifies its column, row, width and height.
# box = 'column row width height'
# They indicate the 1-based column/row to position the top-left corner of the card.

# In this example, we place a 1x1 card in each column/row on a page
# to demonstrate their column/row values.

page = site['/demo']
columns = 12
rows = 10

for column in range(1, columns + 1):
    for row in range(1, rows + 1):
        box = f'{column} {row} 1 1'
        page[f'card_{column}_{row}'] = ui.markdown_card(box=box, title=box, content='')

page.save()
